test/cpp/end2end/generic_end2end_test.cc

```diff
@@ -330,6 +381,32 @@ TEST_F(GenericEnd2endTest, Deadline) {
                        gpr_time_from_seconds(10, GPR_TIMESPAN)));
 }
 
+TEST_F(GenericEnd2endTest, ShortDeadline) {
+  ResetStub();
+
+  ClientContext cli_ctx;
+  EchoRequest request;
+  EchoResponse response;
+
+  {
+    std::lock_guard<std::mutex> lock(shutting_down_mu_);
+    shutting_down_ = false;
+  }
+  std::thread driver([=] { DriveCompletionQueue(); });
```

Comment:
Prefer not to use an anonymous lambda capture; can you specify the things that need to be captured? Is it just "this" ? Alterrnatively, if it's easier, you could do the sync client call in the thread and only capture the things that you need for that.

Reply:
Done, just needed 'this'